libminifi/src/FlowController.cpp

```diff
@@ -163,6 +174,31 @@ FlowController::~FlowController() {
   provenance_repo_ = nullptr;
 }
 
+bool FlowController::applyConfiguration(std::string &configurePayload) {
```

Comment:
I have a general comment. I'm putting it here because as an alternative approach could we put the listener changes into the FlowConfiguration and potentially have a callback. this might get rid of the need for the forward declaration and for the listener to know anything about the flow controller. it would be nice to abstract this into some kind of c2 suite so that we an encapsulate this and everything else into a package that's homogenous.